fix: condense Scoring & Sponsorship section and add maintainer bonus table to readme

RaghavArora14 opened the pull request:
fixes #248
<img width="963" height="906" alt="image" src="https://github.com/user-attachments/assets/03a01304-6582-4741-a76e-7d64db2a0b97" />
<img width="1098" height="544" alt="image" src="https://github.com/user-attachments/assets/1183c017-4cbc-4771-9eb2-3ba9111ec951" />

This update simplifies the Scoring & Sponsorship System in the README by condensing explanations into a shorter, skimmable format while keeping the key tables. It clarifies how weekly scores map to star strings, how monthly sponsorship is calculated from weekly star counts (including median/min/max logic and highScore fallback), and adds a clear table for maintainer bonuses by level. Crowns are explicitly noted as counting as 3 stars for sponsorship.

techmannih reviewed (commented):
I think the issue is related to adding points to the star table — basically, how many points correspond to how many stars. But you can also add the rule like this:
tiny = 1 point = 1 star,
minor = 2 points = 2 stars,
major = 4 points = 3 stars.

techmannih commented:
> the issue also mentioned how sponsorship system works

ohh yes, make sense

RaghavArora14 commented on README.md:
```diff
@@ -138,6 +141,66 @@ pie
+| 31-50 | ⭐⭐⭐ | 3 stars |
+| 51-75 | 👑 | 3 stars |
+| 76-100 | 👑👑 | 3 stars |
+| 101+ | 👑👑👑 | 3 stars |
```
no other mention of this logic except this
<img width="375" height="686" alt="image" src="https://github.com/user-attachments/assets/5c79cf82-68bc-4c81-a3c9-734046695b97" />

RaghavArora14 replied:
other mentions seem to be unrelated

ShiboSoftwareDev replied:
@RaghavArora14 just call them 1/2/3 crown/s

RaghavArora14 replied:
okay and what about the stars? they stay the the same so kind of like this 

| Score Range | Star String | Count Value |
|------------|-------------|-------------|
| 0-3 | (empty) | 0 stars |
| 4-10 | ⭐ | 1 star |
| 11-30 | ⭐⭐ | 2 stars |
| 31-50 | ⭐⭐⭐ | 3 stars |
| 51+ | 1,2,3 👑 | 3 stars |



??
or 



| Score Range | Star String | Count Value |
|------------|-------------|-------------|
| 0-3 | (empty) | 0 stars |
| 4-10 | ⭐ | 1 star |
| 11-30 | ⭐⭐ | 2 stars |
| 31-50 | ⭐⭐⭐ | 3 stars |
| 51-75 | 👑 | 3 stars |
| 76-100 | 2👑 | 3 stars |
| 101+ | 3👑 |  3 stars |

ShiboSoftwareDev replied:
instead of 3 stars for the crown, just say 1 crown, a crown is not just 3 stars it's more than that(even though we handle them similarly currently)

RaghavArora14 replied:
so like this

| Score Range | Star String | Count Value |
|------------|-------------|-------------|
| 0-3 | (empty) | 0 stars |
| 4-10 | ⭐ | 1 star |
| 11-30 | ⭐⭐ | 2 stars |
| 31-50 | ⭐⭐⭐ | 3 stars |
| 51-75 | 1👑 |  |
| 76-100 | 2👑 |  |
| 101+ | 3👑 |  |

ShiboSoftwareDev replied:
just literally say 1 crown to the right of it, that should be clearer, since it's already clear that it's worth at least 3 stars

RaghavArora14 commented:
> wait why is the readme modified? you've got to modify the script that generates the readme

updated both, added updated snapshots, left a comment on discord

RaghavArora14 commented on tests/__snapshots__/test-generate-markdown.test.ts.snap:
```diff
@@ -1,8 +1,15 @@
-// Bun Snapshot v1, https://goo.gl/fbAQLP
+// Bun Snapshot v1, https://bun.sh/docs/test/snapshots
```
Older versions of Bun used goo.gl/fbAQLP (a Google short URL) in snapshot headers
Newer Bun versions (like your 1.3.1) use bun.sh/docs/test/snapshots (direct docs link)
When Bun rewrites snapshots, it uses its current header format
This is completely normal and expected behavior
